src/waveform/renderers/waveformrendermark.cpp

```diff
@@ -146,6 +146,7 @@ void WaveformRenderMark::slotCuesUpdated() {
 
 void WaveformRenderMark::generateMarkImage(WaveformMarkPointer pMark) {
     // Load the pixmap from file -- takes precedence over text.
+    float devicePixelRatio = m_waveformRenderer->getDevicePixelRatio();
     if (!pMark->m_pixmapPath.isEmpty()) {
         QString path = pMark->m_pixmapPath;
         QImage image = *WImageStore::getImage(path, scaleFactor());
```

Comment:
Here scaleFactor() is in use. 
Which value has it?  
I did not understand why scaleFactor() works here. 
Does it? 
When is the code used?

Reply:
scaleFactor or devicePixelRatio makes no difference here.
It's 1.0 regardless of the Qt scale factor. some leftover from custom Mixxx scaling? 
this is used when markers use icons, like in Deere (Cue, loop_in/out).
then the entire part below is skipped
= the icon needs to be centered so the icon stroke (if any) aligns with the marker position on the waveform

also, the one initial problem here causing the offset is that the marker sroke is not taken from the samplePosition but directly.

Reply:
> I did not understand why scaleFactor() works here.

must admit I don't understand it either.
The fixes apply to all QPainter-related calculations below.

Reply:
Deere uses it for its cue marker. 
I can confirm that it scales fine.